### PR TITLE
test: added test cases in cypress

### PR DESCRIPTION
**Pull request by preetamrevankar**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [x] CI/CD

## How did you test it?

Screenshot 2024-11-06 at 4 42 31 PM

## Checklist

- [ ] I ran `npm run re:build`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: cypress-tests/cypress/e2e/stripe-3ds-test.cy.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    getIframeBody().find('[data-testid=cardNoInput]').type(cardNo); // Example card number
+    getIframeBody().find('[data-testid=expiryInput]').type(card_exp_month); // Expiration month
+    getIframeBody().find('[data-testid=expiryInput]').type(card_exp_year); // Expiration year
+    getIframeBody().find('[data-testid=cvvInput]').type(cvc); // CVV
+
+    // Click on the submit button
+    getIframeBody().get("#submit").click();
+
+    // Wait for the response and assert payment success message
+    cy.wait(3000); // Adjust wait time based on actual response time
```
Comment (Sanskar2001) [inline cypress-tests/cypress/e2e/stripe-3ds-test.cy.ts:54]:
```suggestion
     getIframeBody().get("#submit").click().then(() => {
      cy.url().should('include', 'hooks.stripe.com/3d_secure_2');
    });
```

File: cypress-tests/cypress/e2e/stripe-3ds-test.cy.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    cy.url().should('include', 'hooks.stripe.com/3d_secure_2');
```
Comment (Sanskar2001) [inline cypress-tests/cypress/e2e/stripe-3ds-test.cy.ts:55]:
```suggestion
```

File: cypress-tests/cypress/e2e/stripe-Manualcapture-test.cy.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    getIframeBody().find('[data-testid=expiryInput]').type(card_exp_year); // Expiration year
+    getIframeBody().find('[data-testid=cvvInput]').type(cvc); // CVV
+
+    // Click on the submit button
+    getIframeBody().get("#submit").click();
+
+     // Wait for the response and assert payment processing message
+  cy.wait(3000);
+  cy.contains('Payment processing! Requires manual capture')
+  .should('be.visible');
```
Comment (Sanskar2001) [inline cypress-tests/cypress/e2e/stripe-Manualcapture-test.cy.ts:46]:
```suggestion
        getIframeBody().get("#submit").click().then(() => {
      cy.contains('Payment processing! Requires manual capture')
        .should('be.visible', { timeout: 3000 });
    })
```

File: cypress-tests/cypress/e2e/stripe-Manualcapture-test.cy.ts
Comment (Sanskar2001) [inline cypress-tests/cypress/e2e/stripe-Manualcapture-test.cy.ts:1]: rename the spec file to stripe-manual-capture-card-flow-e2e-test.cy.ts

File: cypress-tests/cypress/support/cards.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        card_exp_month:"12",
+        card_exp_year:"30",
+    },
+    invalidMonth: {
+        cardNo: "4000000000000002",
+        cardScheme: "Visa",
+        cvc: "123",
+        card_exp_month:"13",
+        card_exp_year:"30",
+    },
```
Comment (Sanskar2001) [inline cypress-tests/cypress/support/cards.ts:64]:
```suggestion
```

Comment (Sanskar2001) [reply cypress-tests/cypress/support/cards.ts:64]: If you're testing for an invalid card expiry year, you can use any card and simply enter an invalid value in the expiry field. Likewise, you can remove other cards also.

File: cypress-tests/cypress/e2e/Stripe-no3ds-test.cy.ts
Comment (Sanskar2001) [inline cypress-tests/cypress/e2e/Stripe-no3ds-test.cy.ts:1]: Rename this spec file to stripe-no-3DS-card-flow-e2e-test.cy.js

**Review by Sanskar2001 — CHANGES_REQUESTED** [review]

File: cypress-tests/cypress/e2e/stripe-3ds-test.cy.ts (hunk 2 above)
Comment (Sanskar2001) [inline cypress-tests/cypress/e2e/stripe-3ds-test.cy.ts:55]:
```suggestion
    getIframeBody().get("#submit").click().then(() => {
	      cy.url().should('include', 'hooks.stripe.com/3d_secure_2');
	    });
```

File: cypress-tests/cypress/e2e/Stripe-no3ds-test.cy.ts
Comment (preetamrevankar) [reply cypress-tests/cypress/e2e/Stripe-no3ds-test.cy.ts:1]: Done with the changes

File: cypress-tests/cypress/e2e/stripe-3ds-test.cy.ts (hunk 2 above)
Comment (preetamrevankar) [reply cypress-tests/cypress/e2e/stripe-3ds-test.cy.ts]: Done with the changes

File: cypress-tests/cypress/support/cards.ts (hunk 4 above)
Comment (preetamrevankar) [reply cypress-tests/cypress/support/cards.ts]: Done with the changes

File: cypress-tests/cypress/e2e/stripe-Manualcapture-test.cy.ts
Comment (preetamrevankar) [reply cypress-tests/cypress/e2e/stripe-Manualcapture-test.cy.ts:1]: Done.

File: cypress-tests/cypress/e2e/stripe-Manualcapture-test.cy.ts (hunk 3 above)
Comment (preetamrevankar) [reply cypress-tests/cypress/e2e/stripe-Manualcapture-test.cy.ts]: Done.

File: cypress-tests/cypress/e2e/stripe-3ds-test.cy.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+  it("orca-payment-element iframe loaded", () => {
+    cy.get(iframeSelector)
+      .should("be.visible")
+      .its("0.contentDocument")
+      .its("body");
+  });
+
+  it("should complete the card payment successfully", () => {
+    // Visit the page with the payment form
```
[the remark this answers is not in the document]
Comment (preetamrevankar) [reply cypress-tests/cypress/e2e/stripe-3ds-test.cy.ts]: Done.

File: cypress-tests/cypress/e2e/stripe-3ds-test.cy.ts
[the remark this answers is not in the document]
Comment (preetamrevankar) [reply cypress-tests/cypress/e2e/stripe-3ds-test.cy.ts:1]: Done.

File: cypress-tests/cypress/e2e/savedcard-test.cy.ts
[the remark this answers is not in the document]
Comment (preetamrevankar) [reply cypress-tests/cypress/e2e/savedcard-test.cy.ts:1]: Done.

**Review by Sanskar2001 — APPROVED** [review]

**Review by Sanskar2001 — APPROVED** [review]

**Review by PritishBudhiraja — APPROVED** [review]

**Pull request merged**